src/transformers/utils/import_utils.py

```diff
@@ -1581,3 +1655,173 @@ def direct_transformers_import(path: str, file="__init__.py") -> ModuleType:
+                #     backends=(
+                #             "sentencepiece", "tf"
+                #     )
+                # )
+                elif "backends" in lines[previous_index + 1]:
+                    backends = []
+                    for backend_line in lines[previous_index:index]:
+                        if "backends" in backend_line:
+                            backend_line = backend_line.split("=")[1]
+                        if '"' in backend_line or "'" in backend_line:
+                            if ", " in backend_line:
+                                backends.extend(backend.strip("()\"', ") for backend in backend_line.split(", "))
+                            else:
+                                backends.append(backend_line.strip("()\"', "))
+
+                        # If the line is only a ')', then we reached the end of the backends and we break.
+                        if backend_line.strip() == ")":
+                            break
+                    backends = tuple(backends)
+
+                # No backends are registered
+                else:
+                    backends = ()
+
+                backends = frozenset(backends)
+                if backends not in module_requirements:
+                    module_requirements[backends] = {}
+                if module_name not in module_requirements[backends]:
+                    module_requirements[backends][module_name] = []
+
+                if not line.startswith("class") and not line.startswith("def"):
+                    skip_line = True
+                else:
+                    start_index = 6 if line.startswith("class") else 4
+                    object_name = line[start_index:].split("(")[0].strip(":")
+                    module_requirements[backends][module_name].append(object_name)
+
+            if not skip_line:
+                previous_line = line
+                previous_index = index
+
+    import_structure = {**module_requirements, **import_structure}
+    return import_structure
+
+
+def spread_import_structure(nested_import_structure):
+    def propagate_tuple(unordered_import_structure):
+        tuple_first_import_structure = {}
+        for _key, _value in unordered_import_structure.items():
+            if not isinstance(_value, dict):
+                tuple_first_import_structure[_key] = _value
+
+            elif any(isinstance(v, tuple) for v in _value.keys()):
+                # Here we want to switch around key and v
+                for k, v in _value.items():
+                    if isinstance(k, tuple):
+                        if k not in tuple_first_import_structure:
+                            tuple_first_import_structure[k] = {}
+                        tuple_first_import_structure[k][_key] = v
+
+            else:
+                tuple_first_import_structure[_key] = propagate_tuple(_value)
+
+        return tuple_first_import_structure
+
+    def flatten_dict(_dict, previous_key=None):
```

Comment:
hehe this looks very similar to [this](https://github.com/huggingface/transformers/blob/517df566f572d90e6301df87870f651f0d1b1110/utils/diff_model_converter.py#L87-L94) nice